doc/classes/String.xml

```diff
@@ -714,7 +788,7 @@
 			<param index="1" name="allow_empty" type="bool" default="true" />
 			<param index="2" name="maxsplit" type="int" default="0" />
 			<description>
-				Splits the string by a [param delimiter] string and returns an array of the substrings. The [param delimiter] can be of any length. If [param delimiter] is an empty string, each substring will be a single character.
+				Splits the string by a [param delimiter] string and returns an array of the substrings. If [param delimiter] is an empty string, each substring will be a single character.
 				If [param allow_empty] is [code]true[/code], and there are two adjacent delimiters in the string, it will add an empty string to the array of substrings at this position.
 				If [param maxsplit] is specified, it defines the number of splits to do from the left up to [param maxsplit]. The default value of [code]0[/code] means that all items are split.
 				If you need only one element from the array at a specific index, [method get_slice] is a more performant option.
```

Comment:
"by a [param delimiter] string" -> "using a [param delimiter]" (consistent with `get_slice()`)

Could specify that it's a PackedStringArray, like you did in `split_floats()`.

I'd touch up the full description here and in `rsplit()`, something like...

- If [param allow_empty] is [code]true[/code], adjacent delimiters will add an empty string to the [PackedStringArray].
- If [param maxsplit] is greater than [code]0[/code], the number of splits from the left cannot exceed [param maxsplit]

The current ones are just too verbose.

Also, take a stab at the examples below. They are wrong, the order is reversed.